hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataLogRecordReader.java

```diff
@@ -110,7 +110,7 @@ public Map<String, HoodieRecord<HoodieMetadataPayload>> getRecordsByKeys(List<St
       return sortedKeys.stream()
           .map(key -> (HoodieRecord<HoodieMetadataPayload>) allRecords.get(key))
           .filter(Objects::nonNull)
-          .collect(Collectors.toMap(HoodieRecord::getRecordKey, r -> r));
+          .collect(Collectors.toMap(HoodieRecord::getRecordKey, r -> r, (r1, r2) -> r2));
```

Comment:
this will be reverted after the new `BaseTableMetadata#getAllRecordsByKeys` is implemented.